### PR TITLE
Tranquillité, Prospérité & Mimeré

### PR DESCRIPTION
**Pull request by Tastyfish**
A partial port of https://github.com/tgstation/-tg-station/pull/14870
- Some ideas from [this thread](http://nanotrasen.se/phpbb3/viewtopic.php?f=12&t=7283&start=10), although this is mainly to make the mime great again.

## What's new?
- Adds new bananium equivalent for mimes: _Tranquillite!_
 - It can be found on a seperate destroyed shuttle near the bananium one.
 - It can refined, minted into coins, etc.
- One can directly form the mineral sheets into an "invisible wall" which is partially visible, like water, and can be broken down with a crowbar back into the mineral.
- One can also directly form the mineral sheets into a silent floor tile that silences jackboots & clown shoes.
- One can grind tranquillite into the Nothing reagent that the mime had always spawned with.
- The Recitence mech is now buildable.

## The Recitence
- It has received a few subtle tweaks to be more on par with the HONK mech, although it is arguably still less annoying. Time will tell, of course.
- The gun now deals more status effects, but no actual brute damage.
 - It also saps a HONK mech's power and temporarily disables the __Honker__ weapon.
 - Likewise, the HONK's honker saps the power of the Recitence, although to a lesser extent, as it's a large ranged AOE weapon, while the mime must aim.

#### Build Steps
1. Create all of the Recitence parts using the exosuit fabricator.
2. Assemble all of the Recitence parts to the chassis.
3. Reticience Central Control module
4. Reticience Peripherals Control module
5. Reticience Weapon Control and Targeting module
6. Beret
7. Make an audible emote, such as `*fart`, or `*scream`. This will sufficiently confuse the Gods of Mimery.
8. Mime mask
9. Audible emote
10. Suspenders
11. Audible emote
12. S.H.H. "Quietus" Carbine
13. Exosuit Module (Mime RCD Module)

:cl:
rscadd: Mimes now have a new ore, Tranquillite, used to bring peace and quiet to the station.
rscadd: There are also permanent invisible walls and silent floors that can be made from tranquillite.
rscadd: The Recitence mech is now buildable!
tweak: The Recitence mech is now playable!
/:cl:

### CODE REVIEW

File: code/game/mecha/equipment/tools/tools.dm [hunk 1]
```diff
@@ -1112,4 +1112,40 @@
...
+/obj/item/mecha_parts/mecha_equipment/tool/mimercd
+	name = "Mounted MRCD"
+	desc = "An exosuit-mounted Mime Rapid Construction Device. (Can be attached to: Recitence)"
+	icon_state = "mecha_rcd"
+	origin_tech = "materials=4;bluespace=3;magnets=4;powerstorage=4"
+	equip_cooldown = 10
+	energy_drain = 250
+	range = MELEE|RANGED
+
+/obj/item/mecha_parts/mecha_equipment/tool/mimercd/Destroy()
```
Comment (Fox-McCloud) [inline code/game/mecha/equipment/tools/tools.dm]: what's the point of this?

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 2]
```diff
@@ -62,6 +63,11 @@ var/global/list/datum/stack_recipe/bananium_recipes = list ( \
 	new/datum/stack_recipe("bananium grenade casing", /obj/item/weapon/grenade/bananade/casing, 4, on_floor = 1), \
 	)
 
+var/global/list/datum/stack_recipe/mime_recipes = list ( \
+	new/datum/stack_recipe("silent tile", /obj/item/stack/tile/silent, 1, 4, 20), \
+	new/datum/stack_recipe("invisible wall", /obj/structure/barricade/mime, 10, one_per_turf = 1, on_floor = 1, time = 50), \
```
Comment (Fox-McCloud) [inline code/game/objects/items/stacks/sheets/mineral.dm]: I have a feeling this is going to be annoying as hell.

File: code/game/mecha/equipment/tools/tools.dm (hunk 1 above)
Comment (Tastyfish) [inline code/game/mecha/equipment/tools/tools.dm]: Whoops, none.

File: code/game/objects/items/stacks/sheets/mineral.dm (hunk 2 above)
Comment (Tastyfish) [inline code/game/objects/items/stacks/sheets/mineral.dm]: To clarify a) they actually look like standing water, not completely invisible, b) you can crowbar them apart, c) At costing 10 sheets, when there's only 30 in existence, it's possibly the worst waste of the material.

**Comment by Tastyfish** [comment]
Whoops, responded to @Fox-McCloud  and then killed it with a line change.
To clarify with the sheet -> invisible wall, the a) the wall look like standing water, not completely invisible, b) is crowbarable, and damageable like a wooden barrier, and c) at costing 5 when there's only 30 sheets of the material in the world, it's kinda wasteful unless you angered Robotics or something.

File: code/game/mecha/equipment/tools/tools.dm [hunk 3]
```diff
@@ -1112,4 +1112,37 @@
...
+			return 1
+	return 0
+
+/obj/item/mecha_parts/mecha_equipment/tool/mimercd/action(atom/target)
+	if(istype(target, /turf/space/transit))//>implying these are ever made -Sieve
+		return
+	if(!istype(target, /turf) && !istype(target, /obj/machinery/door/airlock))
+		target = get_turf(target)
+	if(!action_checks(target) || get_dist(chassis, target)>3) return
+	playsound(chassis, 'sound/machines/click.ogg', 50, 1)
```
Comment (Crazylemon64) [inline code/game/mecha/equipment/tools/tools.dm]: This seems like an egregious oversight, given who this is for...

File: code/game/mecha/equipment/tools/tools.dm [hunk 4]
```diff
@@ -1112,4 +1112,37 @@
...
+
+	if(istype(target, /turf/simulated/floor))
+		occupant_message("Building Wall...")
+		set_ready_state(0)
+		if(do_after_cooldown(target))
+			new /obj/structure/barricade/mime(target)
+			playsound(target, 'sound/items/Deconstruct.ogg', 50, 1)
```
Comment (Crazylemon64) [inline code/game/mecha/equipment/tools/tools.dm]: Likewise.

**Comment by ZomgPonies** [comment]
+1 This is great. I've always wanted to expand clown/mime rivalry in fun ways, and this definitaly fits the bill.

**Comment by TullyBurnalot** [comment]
Finally, the Mime can rise to the challenge of its rivals. :100:

File: code/game/mecha/mecha_parts.dm [hunk 5]
```diff
@@ -333,93 +381,108 @@
...
-		name = "Circuit board (Odysseus Central Control module)"
-		icon_state = "mainboard"
+/obj/item/weapon/circuitboard/mecha/odysseus/main
+	name = "Circuit board (Odysseus Central Control module)"
+	icon_state = "mainboard"
 
-	pod
-		name = "Circuit board (Space Pod Mainboard)"
-		icon_state = "mainboard"
+/obj/item/weapon/circuitboard/mecha/pod
```
Comment (TheDZD) [inline code/game/mecha/mecha_parts.dm]: Death to the relative pathing!

File: code/modules/mining/mint.dm [hunk 6]
```diff
@@ -132,7 +142,7 @@
 					coinsToProduce--
 					newCoins++
 					src.updateUsrDialog()
-					sleep(5);
```
Comment (TheDZD) [inline code/modules/mining/mint.dm]: Holy fucking shitfuck these had colons to end the line, I don't know who though that the sleeps in specific needed to go against the style of pretty much everything else.

File: code/modules/mining/mint.dm [hunk 7]
```diff
@@ -196,9 +214,9 @@
...
+					sleep(5)
 		icon_state = "coinpress0"
-		processing = 0;
```
Comment (TheDZD) [inline code/modules/mining/mint.dm]: Oh god this had it too.

File: code/modules/mining/money_bag.dm [hunk 8]
```diff
@@ -17,25 +17,28 @@
...
 		if (istype(C,/obj/item/weapon/coin/uranium))
-			amt_uranium++;
+			amt_uranium++
 		if (istype(C,/obj/item/weapon/coin/clown))
-			amt_clown++;
+			amt_clown++
+		if (istype(C,/obj/item/weapon/coin/mime))
+			amt_mime++
 		if (istype(C,/obj/item/weapon/coin/adamantine))
-			amt_adamantine++;
```
Comment (TheDZD) [inline code/modules/mining/money_bag.dm]: *scream

File: code/modules/mining/satchel_ore_boxdm.dm [hunk 9]
```diff
@@ -31,25 +31,28 @@
...
-			amt_iron++;
+			amt_iron++
 		if (istype(C,/obj/item/weapon/ore/silver))
-			amt_silver++;
+			amt_silver++
 		if (istype(C,/obj/item/weapon/ore/gold))
-			amt_gold++;
+			amt_gold++
 		if (istype(C,/obj/item/weapon/ore/uranium))
-			amt_uranium++;
```
Comment (TheDZD) [inline code/modules/mining/satchel_ore_boxdm.dm]: No seriously, who thought that these were needed? Did they transition directly from another programming language without bothering to learn DM syntax?

**Comment by TheDZD** [comment]
:fr:

**Pull request merged**